input/postgres/collection_helper.go

```diff
@@ -0,0 +1,71 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"slices"
+
+	"github.com/pganalyze/collector/config"
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+)
+
+type CollectionHelper struct {
```

Comment:
Add comment explaining what this does

Also rename to "Collection" or "CollectionState"?

Reply:
As we discussed in PR review, it's not really just state, so I would avoid the State suffix. We could also use a synonym of Collection like Lot (as in "a number of units of an article, a single article, or a parcel of articles offered as one item (as in an auction sale)") or Batch (I think there's too much potential for confusion with that one, but throwing it out there).